lib/toml/keyvalue.rb

```diff
@@ -10,7 +10,9 @@ def initialize(key, value)
 
     def assign(hash, symbolize_keys = false)
       @symbolize_keys = symbolize_keys
-      fail ValueOverwriteError if hash[key]
+      if hash.has_key?(key)
+        fail ValueOverwriteError, "Key #{key.inspect} is defined more than once"
```

Comment:
I'd like better to define the message on the error class and fail with an instance of the error.

Something like this
```ruby
ValueOverwriteError.new key if hash.key? key
```

> Note: `has_key?` was [deprecated](http://blade.nagaokaut.ac.jp/cgi-bin/scat.rb/ruby/ruby-core/43765)